src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -1190,4 +1197,34 @@ public InputStream getFileContent(BitbucketSCMFile file) throws IOException, Int
         }
         return content;
     }
+
+    @Restricted(ProtectedExternally.class)
+    protected CloseableHttpResponse executeMethod(CloseableHttpClient client, HttpRequestBase httpMethod) throws IOException, InterruptedException {
```

Comment:
IDK if this really needs to be `protected` and `@Restricted`. This matches the comparable methods in `BitbucketCloudApiClient`, but from a quick look I think they could both just be `private`.

Reply:
The correspond methods in `BitbucketCloudApiClient` are extended in `BitbucketIntegrationClientFactory` for tests. I added a new method for Bitbucket server with a similar purpose in [9ffaec0](https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/581/commits/9ffaec0747720bddc786d6bce7f3c2bccb3b5fa1).